libraries/AdaptiveExpressions/BuiltinFunctions/Optional.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using AdaptiveExpressions.Memory;
+
+namespace AdaptiveExpressions.BuiltinFunctions
+{
+    public class Optional : ExpressionEvaluator
```

Comment:
>Optional [](start = 17, length = 8)

Documentation.  For the MostSpecificSelector, this is a short hand so that instead of having to do A & B and A you can do A & optional(B) to mean the same thing.